lib/User_LDAP.php

```diff
@@ -183,15 +190,19 @@ public function getUsers($search = '', $limit = 10, $offset = 0) {
 	public function userExists($uid) {
 		// check if an LdapEntry has been cached already
 		if ($this->userManager->getCachedEntry($uid) !== null) {
-			return true;
+			//return true;
 		}
-
-		// check if a uid -> dn mapping is in the db
-		$dn = $this->userManager->username2dn($uid);
-		if ($dn === false) {
+		try {
+			$dn = $this->userManager->username2dn($uid);
+			if(!$dn) {
+				throw new \OutOfBoundsException("Cannot find dn for uid: $uid - not in mapping table - maybe never logged in?");
+			}
+			$entry = $this->userManager->getUserEntryByDn($dn);
```

Comment:
If it were so easy ... `$this->userManager->dnExistsOnLDAP($dn)` contains fallback logic to detect dn renames and tries to find the user based on the configured uuid attribute, in case it cannot find the dn.

Reply:
in https://github.com/owncloud/user_ldap/pull/125/commits/7a3e061f492759ee97809d66e1db2e95759279b2#diff-9a63dbcca373f6250bc3efc1fc92330eR241
```php
if(!is_array($this->access->readAttribute($dn, '', $this->getConnection()->ldapUserFilter))) {
```
and https://github.com/owncloud/user_ldap/pull/125/commits/7a3e061f492759ee97809d66e1db2e95759279b2#diff-9a63dbcca373f6250bc3efc1fc92330eR254
```php
				if(!is_array($this->access->readAttribute($newDn, '', $this->getConnection()->ldapUserFilter))) {
```
use executeRead() instead of readAttributes. The filter isn't changed by readAttribue(s), the ranged requests are unnecessary when fetching a single dn. We only need to lowercase the keys before putting it into the user entry ... actually that logic can be moved into the UserEntry ...